### PR TITLE
Add better opengraph

### PR DESCRIPTION
**Pull request by CloCkWeRX**
This implements Part 2 of #816 - adding more descriptive metadata on the major pages; so that when we share a link on facebook etc it can be better understood/provide a better thumbnail.

History is a bit funky, so if we're broadly OK with the changes but want that cleaned up; I'll redo this as one big squashed commit.

### CODE REVIEW

File: app/views/comments/show.html.haml [hunk 1]
```diff
@@ -1,4 +1,12 @@
 = content_for :title, @comment.post.subject
+- content_for :opengraph do
+  <meta property="og:image" content="#{avatar_uri(@comment.post.author, 150)}"/>
+  <meta property="og:image:user_generated" content="true" />
+  <meta property="og:title" content="#{@comment.post.subject}" />
+  <meta property="og:description" content="#{ strip_tags(@comment.post.body).split(' ')[0..20].join(' ') }" />
```
Comment (cesy) [inline app/views/comments/show.html.haml]: Any other risks of this doing something dodgy if the user puts in nasty content? Stripping the tags should solve most things, but let's have a brief think about any other security risks.

**Comment by cesy** [comment]
We'll need to have a think about the original_uri failing on Travis, since that may just be an oddity of the way we're testing, but would be good to have a catch for it.

The broad principles of this look good, though I've just done a quick skim.

File: app/views/comments/show.html.haml (hunk 1 above)
Comment (CloCkWeRX) [inline app/views/comments/show.html.haml]: I did think about it at the time - users putting in " are probably going to have a bad time of it; unfortunately. I'll probably go swap to the ```tag()``` method to clean this up in a bit.

**Comment by CloCkWeRX** [comment]
One for me to fix up later:
```
Image Too Small
Provided og:image URL, http://staging.growstuff.org/assets/growstuff-apple-touch-icon-precomposed-86d9827705d880da540a437dd6207bfd5a827761c04ee903d46c2985e33c992d.png was not valid because it did not meet the minimum size constraint of 200px by 200px.
```

... which is probably the cause of the facebook thumbnail image being so inconsistent

**Comment by pozorvlak** [comment]
Looks good! Please squash the history and fix the image-size issue you noticed.

**Pull request closed without merging**